src/LayerEditor/ui/data/regions_model.py

```diff
@@ -96,9 +97,11 @@ def is_region_used(self, reg):
 
     def save(self):
         regions_data = []
-        id_to_idx = {}
         for idx, region in enumerate(self.regions.values()):
-            regions_data.append(region.convert_to_data())
-            id_to_idx[region.id] = idx
-        return (regions_data, id_to_idx)
+            regions_data.append(region.save())
+            region.index = idx
+        return regions_data
 
+    def clear_indexing(self):
```

Comment:
After moving inappropriate code to the le_model (probably). We get again very simple container with add and delete undoable methods. Possibly no need to have special class for that.

Reply:
Even after removing inappropriate code there is still 5 methods. Granted they are simple and it might be possible to merge them into a base class. The code would become unreadable. Also getting rid of those classes would mean moving the code to LEModel, which already has nearly 600 lines of code. Again I think this structure is more readable and managable.